src/test/java/uk/gov/pay/api/resources/directdebit/DirectDebitPaymentsResourceTest.java

```diff
@@ -0,0 +1,152 @@
+            config("connectorDDUrl", "http://localhost:" + CONNECTOR_DD_PORT),
+            config("publicAuthUrl", "http://localhost:" + PUBLIC_AUTH_PORT + "/v1/auth"),
+            config("redis.endpoint", redisDockerRule.getRedisUrl())
+    );
+
+    private ConnectorDDMockClient connectorDDMockClient = new ConnectorDDMockClient(connectorDDMock);
+    private PublicAuthMockClient publicAuthMockClient = new PublicAuthMockClient(publicAuthMock);
+    
+    @Before
+    public void setup() {
+        configuration = app.getConfiguration();
+        connectorDDMock.resetAll();
+        publicAuthMock.resetAll();
+    }
+
+    String paymentEventsLocationFor(String chargeId) {
+        return configuration.getBaseUrl() + "v1/directdebit/payments/" + chargeId + "/events";
+    }
+
+    private String paymentLocationFor(String chargeId) {
+        return configuration.getBaseUrl() + "v1/directdebit/payments/" + chargeId;
+    }
+
+    String mandateLocationFor(String mandateId) {
+        return configuration.getBaseUrl() + "v1/directdebit/mandates/" + mandateId;
+    }
+
+    protected ValidatableResponse postPaymentResponse(CreateDirectDebitPaymentRequest payload) {
+        return given().port(app.getLocalPort())
+                .body(payload)
+                .accept(JSON)
+                .contentType(JSON)
+                .header(AUTHORIZATION, "Bearer " + API_KEY)
+                .post(PAYMENTS_PATH)
+                .then();
+    }
+
+    @Test
+    public void paymentCreatedSuccessfully() throws JsonProcessingException {
+        final Account account = new Account("test", TokenPaymentType.DIRECT_DEBIT);
+        final String paymentId = "abc123";
+        final String createdDate = "2018-01-01T11:12:13Z";
+        final String referenceText = "a reference";
+        final String descriptionText = "a description";
+        final String paymentProviderText = "a payment provider";
+        final String status = "created";
+        final boolean finished = false;
+
+        DirectDebitConnectorCreatePaymentResponse connectorResponse = aDirectDebitConnectorCreatePaymentResponse()
+                .withPaymentExternalId(paymentId)
+                .withAmount(500L)
+                .withPaymentProvider(paymentProviderText)
+                .withCreatedDate(createdDate)
+                .withDescription(descriptionText)
+                .withState(new PaymentState(status, finished))
+                .withReference(referenceText)
+                .build();
+
+        publicAuthMockClient.mapBearerTokenToAccountId(API_KEY, account.getAccountId());
+        connectorDDMockClient.respondWithPaymentCreated(connectorResponse, account.getAccountId());
+
+        var createDirectDebitPaymentRequest = (CreateDirectDebitPaymentRequest) CreatePaymentRequestBuilder
+                .builder()
+                .amount(500)
+                .description(descriptionText)
+                .reference(referenceText)
+                .agreementId("test")
```

Comment:
can this be `mandateId` please?

Reply:
this comment should be addressed...i think

Reply:
ahh yeh sorry, missed this one